README.txt

mk_listbydate is a very simple TYPO3 extension that allows to display a list
of entries by date. The entries must be edited in the backend list-view.

The recurring dates can be set as:
 * Day of month e.g. the 11th of every month
 * e.g. Every monday of the first week of the month or the last Sunday of the month



ChangeLog
2017-03-13 Mattanja Kern <mattanja.kern@kernetics.de>
	* Update for TYPO3 6.2 compatibility

2010-05-15 Mattanja Kern <mattanja.kern@kernetics.de>

	* initial code generated with kickstarter
	* Publish first version


tca.php
<?php
if (!defined ('TYPO3_MODE')) 	die ('Access denied.');

$TCA['tx_mklistbydate_entries'] = array (
	'ctrl' => $TCA['tx_mklistbydate_entries']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => 'hidden,starttime,endtime,text,weekdays,occurrence,dayofmonth'
	),
	'feInterface' => $TCA['tx_mklistbydate_entries']['feInterface'],
	'columns' => array (
		'hidden' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.hidden',
			'config'  => array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'starttime' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.starttime',
			'config'  => array (
				'type'     => 'input',
				'size'     => '8',
				'max'      => '20',
				'eval'     => 'date',
				'default'  => '0',
				'checkbox' => '0'
			)
		),
		'endtime' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.endtime',
			'config'  => array (
				'type'     => 'input',
				'size'     => '8',
				'max'      => '20',
				'eval'     => 'date',
				'checkbox' => '0',
				'default'  => '0',
				'range'    => array (
					'upper' => mktime(3, 14, 7, 1, 19, 2038),
					'lower' => mktime(0, 0, 0, date('m')-1, date('d'), date('Y'))
				)
			)
		),
		'text' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.text',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',	
				'eval' => 'required',
			)
		),
		'weekdays' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.weekdays',		
			'config' => array (
				'type' => 'check',
				'cols' => 4,
				'items' => array (
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.weekdays.I.0', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.weekdays.I.1', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.weekdays.I.2', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.weekdays.I.3', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.weekdays.I.4', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.weekdays.I.5', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.weekdays.I.6', ''),
				),
			)
		),
		'occurrence' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.occurrence',		
			'config' => array (
				'type' => 'check',
				'cols' => 4,
				'items' => array (
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.occurrence.I.0', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.occurrence.I.1', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.occurrence.I.2', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.occurrence.I.3', ''),
					array('LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.occurrence.I.4', ''),
				),
			)
		),
		'dayofmonth' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries.dayofmonth',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '31',
					'lower' => '1'
				),
				'default' => 1
			)
		),
	),
	'types' => array (
		'0' => array('showitem' => 'hidden;;1;;1-1-1, text, weekdays, occurrence, dayofmonth')
	),
	'palettes' => array (
		'1' => array('showitem' => 'starttime, endtime')
	)
);
?>

ext_tables.php
<?php
if (!defined ('TYPO3_MODE')) {
	die ('Access denied.');
}

t3lib_extMgm::allowTableOnStandardPages('tx_mklistbydate_entries');

$TCA['tx_mklistbydate_entries'] = array (
	'ctrl' => array (
		'title'     => 'LLL:EXT:mk_listbydate/locallang_db.xml:tx_mklistbydate_entries',		
		'label'     => 'text',	
		'tstamp'    => 'tstamp',
		'crdate'    => 'crdate',
		'cruser_id' => 'cruser_id',
		'default_sortby' => 'ORDER BY text',	
		'delete' => 'deleted',	
		'enablecolumns' => array (		
			'disabled' => 'hidden',	
			'starttime' => 'starttime',	
			'endtime' => 'endtime',
		),
		'dynamicConfigFile' => t3lib_extMgm::extPath($_EXTKEY).'tca.php',
		'iconfile'          => t3lib_extMgm::extRelPath($_EXTKEY).'icon_tx_mklistbydate_entries.gif',
	),
);

t3lib_div::loadTCA('tt_content');
$TCA['tt_content']['types']['list']['subtypes_excludelist'][$_EXTKEY.'_pi1']='layout,select_key';

// you add pi_flexform to be renderd when your plugin is shown
$TCA['tt_content']['types']['list']['subtypes_addlist'][$_EXTKEY.'_pi1']='pi_flexform';

t3lib_extMgm::addPlugin(array(
	'LLL:EXT:mk_listbydate/locallang_db.xml:tt_content.list_type_pi1',
	$_EXTKEY . '_pi1',
	t3lib_extMgm::extRelPath($_EXTKEY) . 'ext_icon.gif'
),'list_type');


t3lib_extMgm::addStaticFile($_EXTKEY,'pi1/static/','List by date');

// now, add your flexform xml-file
t3lib_extMgm::addPiFlexFormValue($_EXTKEY.'_pi1', 'FILE:EXT:'.$_EXTKEY.'/flexform_ds_pi1.xml');

if (TYPO3_MODE == 'BE') {
	$TBE_MODULES_EXT['xMOD_db_new_content_el']['addElClasses']['tx_mklistbydate_pi1_wizicon'] = t3lib_extMgm::extPath($_EXTKEY).'pi1/class.tx_mklistbydate_pi1_wizicon.php';
}
?>

pi1/class.tx_mklistbydate_pi1.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2010 Mattanja Kern <mattanja.kern@kernetics.de>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 * Hint: use extdeveval to insert/update function index above.
 */

// Fix for TYPO3 >6.2 https://forge.typo3.org/issues/54144
if (!class_exists('tslib_pibase')) require_once(PATH_tslib . 'class.tslib_pibase.php');

/**
 * Plugin 'List by date' for the 'mk_listbydate' extension.
 *
 * @author	Mattanja Kern <mattanja.kern@kernetics.de>
 * @package	TYPO3
 * @subpackage	tx_mklistbydate
 */
class tx_mklistbydate_pi1 extends tslib_pibase {
	var $prefixId      = 'tx_mklistbydate_pi1';		// Same as class name
	var $scriptRelPath = 'pi1/class.tx_mklistbydate_pi1.php';	// Path to this script relative to the extension dir.
	var $extKey        = 'mk_listbydate';	// The extension key.
	var $pi_checkCHash = true;

	var $weekdays = array(
		1 => 'Monday',
		2 => 'Tuesday',
		4 => 'Wednesday',
		8 => 'Thursday',
		16 => 'Friday',
		32 => 'Saturday',
		64 => 'Sunday'
	);

	var $binaryLastRecurrence = 16;

	/**
	 * Occurrances:
	 * 1 - first
	 * 2 - second
	 * 4 - third
	 * 8 - fourth
	 * 16 - last
	 */ 

	/**
	 * Main method of your PlugIn
	 *
	 * @param	string		$content: The content of the PlugIn
	 * @param	array		$conf: The PlugIn Configuration
	 * @return	The content that should be displayed on the website
	 */
	function main($content, $conf)	{
		$this->init();
		switch($this->lConf['displayType']) {
			case 1: // Montly
				if (strstr($this->cObj->currentRecord,'tt_content'))	{
					$conf['pidList'] = $this->cObj->data['pages'];
					$conf['recursive'] = $this->cObj->data['recursive'];
				}
				return $this->pi_wrapInBaseClass($this->monthListView($content, $conf));
			case 2: // Weekly
				if (strstr($this->cObj->currentRecord,'tt_content'))	{
					$conf['pidList'] = $this->cObj->data['pages'];
					$conf['recursive'] = $this->cObj->data['recursive'];
				}
				return $this->pi_wrapInBaseClass($this->weekListView($content, $conf));
			case 3: // Show all
				if (strstr($this->cObj->currentRecord,'tt_content'))	{
					$conf['pidList'] = $this->cObj->data['pages'];
					$conf['recursive'] = $this->cObj->data['recursive'];
				}
				return $this->pi_wrapInBaseClass($this->showAllView($content, $conf));
			default: // 0 (Simple list view)
				if (strstr($this->cObj->currentRecord,'tt_content'))	{
					$conf['pidList'] = $this->cObj->data['pages'];
					$conf['recursive'] = $this->cObj->data['recursive'];
				}
				return $this->pi_wrapInBaseClass($this->simpleView($content, $conf));
		}
	}

	function init() {
		$this->pi_initPIflexForm(); // Init and get the flexform data of the plugin
		$this->lConf = array(); // Setup our storage array...
		// Assign the flexform data to a local variable for easier access
		$piFlexForm = $this->cObj->data['pi_flexform'];
		// Traverse the entire array based on the language...
		// and assign each configuration option to $this->lConf array...
		if (is_array($piFlexForm['data'])) {
		foreach ( $piFlexForm['data'] as $sheet => $data ) {
			if (is_array($data)) {
			foreach ( $data as $lang => $value ) {
				if (is_array($value)) {
				foreach ( $value as $key => $val ) {
					$this->lConf[$key] = $this->pi_getFFvalue($piFlexForm, $key, $sheet);
				}
				}
			}
			}
		}
		}
	}

	function getWeekdayInMonthInfo($timestamp=null) {
		if ($timestamp == null) {
			// Default to current time
			$timestamp = time();
		}
		$currentWeekday = date('N',$timestamp); // 1 (for Monday) through 7 (for Sunday)
		$currentDayofmonth = date('j',$timestamp);
		$recurrence = 0;

		// Weekday of first day in this month
		$tmpWeekday = date('N', mktime(12,0,0,date('n',$timestamp),1,date('Y',$timestamp)));
		$firstOccurrence = ($currentWeekday - $tmpWeekday) + 1;
		if ($firstOccurrence < 1) $firstOccurrence += 7;

		while ($currentDayofmonth >= ($firstOccurrence + ($recurrence * 7))) {
			$recurrence++;
		}

		// Next recurrence would be greater than number of days in the given month
		$isLast = (bool)(($firstOccurrence + (($recurrence+1) * 7)) > date('t',$timestamp));

		return array('recurrence' => $recurrence,
					 'weekday' => $currentWeekday,
					 'isLast' => $isLast);
	}

	function simpleView($content, $conf) {
		$this->conf = $conf;		// Setting the TypoScript passed to this function in $this->conf
		$this->pi_setPiVarDefaults();
		$this->pi_loadLL();		// Loading the LOCAL_LANG values

		if (!isset($this->piVars['pointer']))	$this->piVars['pointer']=0;
		if (!isset($this->piVars['mode']))	$this->piVars['mode']=1;

		// Initializing the query parameters:
		list($this->internal['orderBy'],$this->internal['descFlag']) = explode(':',$this->piVars['sort']);
		$this->internal['results_at_a_time']=t3lib_utility_Math::forceIntegerInRange($lConf['results_at_a_time'],0,1000,10000);		// Number of results to show in a listing.
		$this->internal['maxPages']=t3lib_utility_Math::forceIntegerInRange($lConf['maxPages'],0,1000,2);;		// The maximum number of "pages" in the browse-box: "Page 1", "Page 2", etc.
		$this->internal['searchFieldList']='text';
		$this->internal['orderByList']='uid,text';

		/**
		 * Filter by query (instead of loop)
		 */
		$weekdayinfo = $this->getWeekdayInMonthInfo();
		$binaryWeekday = pow(2,($weekdayinfo['weekday']-1));
		$binaryRecurrence = pow(2,$weekdayinfo['recurrence']-1);
		
		// Build query
		$where = " AND (";
		// Weekday
		$where .= "(".$binaryWeekday."=(".$binaryWeekday."&weekdays)) ";
		// Week of month
		$where .= " AND (((".$binaryRecurrence."=(".$binaryRecurrence."&occurrence)))";
		if ($weekdayinfo['isLast']) {
			$where .= " OR (16=(16&occurrence))";
		}
		$where .= ")";
		// Day of month
		$where .= " OR (dayofmonth=".date('j').")";
		$where .= ")";
		// Get number of records:
		$res = $this->pi_exec_query('tx_mklistbydate_entries',1, $where);
		list($this->internal['res_count']) = $GLOBALS['TYPO3_DB']->sql_fetch_row($res);

		// Make listing query, pass query to SQL database:
		$res = $this->pi_exec_query('tx_mklistbydate_entries', 0, $where);
		$this->internal['currentTable'] = 'tx_mklistbydate_entries';

		// Put the whole list together:
		$fullTable='';	// Clear var;

		// Custom output instead of default table
		//$fullTable.=$this->pi_list_makelist($res);

		// Make list table header:
		$tRows=array();
		$this->internal['currentRow']='';
		//$tRows[] = $this->pi_list_header();

		// Make list table rows
		while($this->internal['currentRow'] = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($res)) {
			$tRows[] = '<li>'.($this->getFieldContent('text')).'</li>';
		}

		$out = '<div'.$this->pi_classParam('listrow').'>
			<ul>
			'.implode('',$tRows).'
			</ul>
		</div>';
		return $out;
	}

	function showAllView($content, $conf) {
		$this->conf = $conf;
		$this->pi_setPiVarDefaults();
		$this->pi_loadLL();	// Loading the LOCAL_LANG values

		if (!isset($this->piVars['pointer']))	$this->piVars['pointer']=0;
		if (!isset($this->piVars['mode']))	$this->piVars['mode']=1;

		// Initializing the query parameters:
		list($this->internal['orderBy'],$this->internal['descFlag']) = explode(':',$this->piVars['sort']);
		$this->internal['results_at_a_time']=t3lib_utility_Math::forceIntegerInRange($lConf['results_at_a_time'],0,1000,10000);
		$this->internal['maxPages']=t3lib_utility_Math::forceIntegerInRange($lConf['maxPages'],0,1000,2);
		$this->internal['searchFieldList']='text';
		$this->internal['orderByList'] = 'uid,text';

		// Build query
		$where = '';
		$cat = '';
		$groupBy = 'text';
		$orderBy = 'text';

		$res = $this->pi_exec_query('tx_mklistbydate_entries',1, $where, $cat, $groupBy, $orderBy);
		list($this->internal['res_count']) = $GLOBALS['TYPO3_DB']->sql_fetch_row($res);

		// Make listing query, pass query to SQL database:
		$res = $this->pi_exec_query('tx_mklistbydate_entries', 0, $where, $cat, $groupBy, $orderBy);
		$this->internal['currentTable'] = 'tx_mklistbydate_entries';

		// Make list table header:
		$tRows=array();
		$this->internal['currentRow']='';
		//$tRows[] = $this->pi_list_header();

		// Make list table rows
		while($this->internal['currentRow'] = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($res)) {
			$tRows[] = '<li>'.($this->getFieldContent('text')).'</li>';
		}

		$out = '<div'.$this->pi_classParam('listrow').'>
			<ul>
			'.implode('',$tRows).'
			</ul>
		</div>';
		return $out;
	}


	function monthListView($content, $conf) {
		$this->conf = $conf;
		$this->pi_setPiVarDefaults();
		$this->pi_loadLL();

		if (!isset($this->piVars['pointer']))	$this->piVars['pointer']=0;
		if (!isset($this->piVars['mode']))	$this->piVars['mode']=1;

		// Initializing the query parameters:
		list($this->internal['orderBy'],$this->internal['descFlag']) = explode(':',$this->piVars['sort']);
		$this->internal['results_at_a_time']=t3lib_utility_Math::forceIntegerInRange($lConf['results_at_a_time'],0,10000,10000); // Number of results to show in a listing.
		$this->internal['maxPages']=t3lib_utility_Math::forceIntegerInRange($lConf['maxPages'],0,1000,2); // The maximum number of "pages" in the browse-box: "Page 1", "Page 2", etc.
		$this->internal['searchFieldList']='text';
		$this->internal['orderByList']='uid,text';

		// Build query
		$where = " AND (dayofmonth>0)";
		$cat = '';
		$groupBy = '';
		$orderBy = 'dayofmonth,text';

		// Get number of records:
		$res = $this->pi_exec_query('tx_mklistbydate_entries',1, $where, $cat, $groupBy, $orderBy);
		list($this->internal['res_count']) = $GLOBALS['TYPO3_DB']->sql_fetch_row($res);

		// Make listing query, pass query to SQL database:
		$res = $this->pi_exec_query('tx_mklistbydate_entries', 0, $where, $cat, $groupBy, $orderBy);
		$this->internal['currentTable'] = 'tx_mklistbydate_entries';

		// Make list table header:
		$tRows=array();
		$this->internal['currentRow']='';

		// Make list table rows
		$cday = 0;
		while($this->internal['currentRow'] = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($res)) {
			$tmp = $this->getFieldContent('dayofmonth');
			if ($tmp == $cday) {
				$text = '';
			} else {
				// Next day
				$cday = $tmp;
				$text = $tmp.".";
			}
			$tRows[] = '<tr><td>'.$text.'</td><td>'.$this->getFieldContent('text').'</td></tr>';
			$c++;
		}

		// table class default: '.$this->pi_classParam('listrow').'
		$out = '<div class="mklistbydate_MonthList"><table class="mklistbydate_MonthList">
			<tbody>
			'.implode('',$tRows).'
			</tbody>
		</table></div>';
		return $out;
	}

	function weekListView($content, $conf) {
		$this->conf = $conf;
		$this->pi_setPiVarDefaults();
		$this->pi_loadLL();

		if (!isset($this->piVars['pointer']))	$this->piVars['pointer']=0;
		if (!isset($this->piVars['mode']))	$this->piVars['mode']=1;

		// Initializing the query parameters:
		list($this->internal['orderBy'],$this->internal['descFlag']) = explode(':',$this->piVars['sort']);
		$this->internal['results_at_a_time']=t3lib_utility_Math::forceIntegerInRange($lConf['results_at_a_time'],0,10000,10000); // Number of results to show in a listing.
		$this->internal['maxPages']=t3lib_utility_Math::forceIntegerInRange($lConf['maxPages'],0,1000,2); // The maximum number of "pages" in the browse-box: "Page 1", "Page 2", etc.
		$this->internal['searchFieldList']='text';
		$this->internal['orderByList']='uid,text';

		// Build query
		$where = " AND weekdays<>0 AND occurrence<>0";
		$cat = '';
		$groupBy = '';
		$orderBy = 'text';

		// Get number of records:
		$res = $this->pi_exec_query('tx_mklistbydate_entries', 1, $where, $cat, $groupBy, $orderBy);
		list($this->internal['res_count']) = $GLOBALS['TYPO3_DB']->sql_fetch_row($res);

		// Make listing query, pass query to SQL database:
		$res = $this->pi_exec_query('tx_mklistbydate_entries', 0, $where, $cat, $groupBy, $orderBy);
		$this->internal['currentTable'] = 'tx_mklistbydate_entries';

		// Make list table header:
		$tRows=array();
		$this->internal['currentRow']='';

		// Make list table rows
		$data = array();
		while($this->internal['currentRow'] = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($res))
		{
			$text = $this->getFieldContent('text');
			$weekdays = $this->getFieldContent('weekdays');
			$occurrence = $this->getFieldContent('occurrence');

			for ($week=1; $week<=16; $week=$week*2)
			{
				if (((int)$week & $occurrence) == $week)
				{
					// Loop the weekdays
					for ($weekday = 1; $weekday<=64; $weekday=$weekday*2)
					{
						if (((int)$weekday & $weekdays) == $weekday)
						{
							$data[$week][$weekday][] = $text;
						}
					}
				}
			}
		}

		$out = '<div class="mklistbydate_WeekList"><table class="mklistbydate_WeekListView">';
		// Loop the weeks
		$llw = 0;
		for ($week=1; $week<=16; $week=$week*2)
		{
			$out .= '<tr><td class="week">'.$this->pi_getLL('tx_mklistbydate_entries.occurrence.I.'.$llw++).'</td><td><table class="mklistbydate_Days">';
			// Loop the weekdays
			$lld = 0;
			for ($weekday = 1; $weekday<=64; $weekday=$weekday*2)
			{
				$out .= '<tr><td class="day">'.$this->pi_getLL('tx_mklistbydate_entries.weekdays.I.'.$lld++).'</td><td><table class="mklistbydate_Items">';
				
				foreach ($data[$week][$weekday] as $item)
				{
					$out .= '<tr><td class="item">'.$item.'</td></tr>';
				}
				
				$out .= '</table></td></tr>';
			}
			$out .= '</table></td></tr>';
		}
		$out .= '</table></div>';
		return $out;
	}

	/**
	 * Returns the content of a given field
	 *
	 * @param	string		$fN: name of table field
	 * @return	Value of the field
	 */
	function getFieldContent($fN)	{
		switch($fN) {
			case 'uid':
				return $this->pi_list_linkSingle($this->internal['currentRow'][$fN],$this->internal['currentRow']['uid'],1);	// The "1" means that the display of single items is CACHED! Set to zero to disable caching.
			break;
			
			default:
				return $this->internal['currentRow'][$fN];
			break;
		}
	}

	/**
	 * Returns the label for a fieldname from local language array
	 *
	 * @param	[type]		$fN: ...
	 * @return	[type]		...
	 */
	function getFieldHeader($fN)	{
		switch($fN) {
			
			default:
				return $this->pi_getLL('listFieldHeader_'.$fN,'['.$fN.']');
			break;
		}
	}
	
	/**
	 * Returns a sorting link for a column header
	 *
	 * @param	string		$fN: Fieldname
	 * @return	The fieldlabel wrapped in link that contains sorting vars
	 */
	function getFieldHeader_sortLink($fN)	{
		return $this->pi_linkTP_keepPIvars($this->getFieldHeader($fN),array('sort'=>$fN.':'.($this->internal['descFlag']?0:1)));
	}
}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/mk_listbydate/pi1/class.tx_mklistbydate_pi1.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/mk_listbydate/pi1/class.tx_mklistbydate_pi1.php']);
}

?>

pi1/class.tx_mklistbydate_pi1_wizicon.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2010 Mattanja Kern <mattanja.kern@kernetics.de>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 * Hint: use extdeveval to insert/update function index above.
 */




/**
 * Class that adds the wizard icon.
 *
 * @author	Mattanja Kern <mattanja.kern@kernetics.de>
 * @package	TYPO3
 * @subpackage	tx_mklistbydate
 */
class tx_mklistbydate_pi1_wizicon {

					/**
					 * Processing the wizard items array
					 *
					 * @param	array		$wizardItems: The wizard items
					 * @return	Modified array with wizard items
					 */
					function proc($wizardItems)	{
						global $LANG;

						$LL = $this->includeLocalLang();

						$wizardItems['plugins_tx_mklistbydate_pi1'] = array(
							'icon'=>t3lib_extMgm::extRelPath('mk_listbydate').'pi1/ce_wiz.gif',
							'title'=>$LANG->getLLL('pi1_title',$LL),
							'description'=>$LANG->getLLL('pi1_plus_wiz_description',$LL),
							'params'=>'&defVals[tt_content][CType]=list&defVals[tt_content][list_type]=mk_listbydate_pi1'
						);

						return $wizardItems;
					}

					/**
					 * Reads the [extDir]/locallang.xml and returns the $LOCAL_LANG array found in that file.
					 *
					 * @return	The array with language labels
					 */
					function includeLocalLang()	{
						$llFile = t3lib_extMgm::extPath('mk_listbydate').'locallang.xml';
						$l10nparser = t3lib_div::makeInstance('t3lib_l10n_parser_Llxml');
						$LOCAL_LANG = $l10nparser->getParsedData($llFile, $GLOBALS['LANG']->lang);

						return $LOCAL_LANG;
					}
				}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/mk_listbydate/pi1/class.tx_mklistbydate_pi1_wizicon.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/mk_listbydate/pi1/class.tx_mklistbydate_pi1_wizicon.php']);
}

?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "mk_listbydate".
#
# Auto generated 15-05-2010 01:06
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'List by Date',
	'description' => 'A list with simple text entries that can be filtered by date',
	'category' => 'fe',
	'author' => 'Mattanja Kern',
	'author_email' => 'mattanja.kern@kernetics.de',
	'shy' => '',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'beta',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 1,
	'lockType' => '',
	'author_company' => '',
	'version' => '1.0.0',
	'constraints' => array(
		'depends' => array(
			'typo3' => '4.5.0-6.2.99'
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'suggests' => array(
	),
	'_md5_values_when_last_written' => 'a:18:{s:9:"ChangeLog";s:4:"a63a";s:10:"README.txt";s:4:"6120";s:12:"ext_icon.gif";s:4:"1bdc";s:17:"ext_localconf.php";s:4:"50f6";s:14:"ext_tables.php";s:4:"6c11";s:14:"ext_tables.sql";s:4:"14e4";s:32:"icon_tx_mklistbydate_entries.gif";s:4:"401f";s:13:"locallang.xml";s:4:"9706";s:16:"locallang_db.xml";s:4:"8e03";s:7:"tca.php";s:4:"ec57";s:19:"doc/wizard_form.dat";s:4:"b6e3";s:20:"doc/wizard_form.html";s:4:"7fd3";s:14:"pi1/ce_wiz.gif";s:4:"02b6";s:33:"pi1/class.tx_mklistbydate_pi1.php";s:4:"136d";s:41:"pi1/class.tx_mklistbydate_pi1_wizicon.php";s:4:"7f8d";s:13:"pi1/clear.gif";s:4:"cc11";s:17:"pi1/locallang.xml";s:4:"a170";s:20:"pi1/static/setup.txt";s:4:"ebab";}',
);

?>